### PR TITLE
feat(provider/azure): Add redblack strategy for azure

### PR DESCRIPTION
**Pull request by neil-yechenwei**
Background:
Add redblack strategy for azure provider.
There is another related PR (https://github.com/spinnaker/deck/pull/6801) with same name in deck for azure redblack strategy. Please also help to review.

### CODE REVIEW

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/model/ResizeAzureServerGroupDescription.groovy [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2014 Netflix, Inc.
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/model/ResizeAzureServerGroupDescription.groovy]: s/2014/2019
s/Netflix, Inc./The original authors

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/ResizeAzureServerGroupAtomicOperation.groovy [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+package com.netflix.spinnaker.clouddriver.azure.resources.servergroup.ops
+
+import com.netflix.frigga.Names
+import com.netflix.spinnaker.clouddriver.azure.resources.servergroup.model.AzureServerGroupDescription
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
+import com.netflix.spinnaker.clouddriver.azure.common.AzureUtilities
+import com.netflix.spinnaker.clouddriver.azure.resources.servergroup.model.ResizeAzureServerGroupDescription
+import com.netflix.spinnaker.clouddriver.azure.resources.cluster.view.AzureClusterProvider
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/ResizeAzureServerGroupAtomicOperation.groovy]: Sort order is off here.

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/converters/ResizeAzureServerGroupAtomicOperationConverter.groovy [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import com.netflix.spinnaker.clouddriver.security.AbstractAtomicOperationsCredentialsSupport
+import groovy.util.logging.Slf4j
+import org.springframework.stereotype.Component
+
+@Slf4j
+@AzureOperation(AtomicOperations.RESIZE_SERVER_GROUP)
+@Component("resizeAzureServerGroupDescription")
+class ResizeAzureServerGroupAtomicOperationConverter extends AbstractAtomicOperationsCredentialsSupport{
+  ResizeAzureServerGroupAtomicOperationConverter() {
+    log.info("Constructor....ResizeAzureServerGroupAtomicOperationConverter")
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/converters/ResizeAzureServerGroupAtomicOperationConverter.groovy]: Extraneous instrumentation?

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/converters/ResizeAzureServerGroupAtomicOperationConverter.groovy [hunk 4]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2016 The original authors.
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/converters/ResizeAzureServerGroupAtomicOperationConverter.groovy]: s/2016/2019

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/ResizeAzureServerGroupAtomicOperation.groovy [hunk 5]
```diff
@@ -0,0 +1,100 @@
+/*
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/ResizeAzureServerGroupAtomicOperation.groovy]: We'd prefer newly-added files to not be authored in groovy. The java files can still live in the same directories as the pre-existing groovy files.

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/model/ResizeAzureServerGroupDescription.groovy (hunk 1 above)
Comment (neil-yechenwei) [reply clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/model/ResizeAzureServerGroupDescription.groovy]: Thanks. Updated to "Copyright 2019 The original authors."

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/ResizeAzureServerGroupAtomicOperation.groovy (hunk 2 above)
Comment (neil-yechenwei) [reply clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/ResizeAzureServerGroupAtomicOperation.groovy]: Thanks. It has been updated.

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/converters/ResizeAzureServerGroupAtomicOperationConverter.groovy (hunk 3 above)
Comment (neil-yechenwei) [reply clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/converters/ResizeAzureServerGroupAtomicOperationConverter.groovy]: Add this here for two reasons.
1. It's good to know what converter is called while running for troubleshooting.
2. Following previous code style like "EnableAzureServerGroupAtomicOperation".

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/converters/ResizeAzureServerGroupAtomicOperationConverter.groovy (hunk 4 above)
Comment (neil-yechenwei) [reply clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/converters/ResizeAzureServerGroupAtomicOperationConverter.groovy]: Thanks. Updated to 2019.

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/ResizeAzureServerGroupAtomicOperation.groovy (hunk 5 above)
Comment (neil-yechenwei) [reply clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/ResizeAzureServerGroupAtomicOperation.groovy]: Thanks for your reminder. As @mybayern1974 said, is it ok to merge it now given one of our customers only use the Red/Black strategy so we'd like to unblock them asap. Meanwhile we will start the work of converting these newly added files to react from now on, and please expect our next PR to have it.

**Comment by duftler** [comment]
Ready to merge once the failing test is addressed.

**Pull request merged**